lib/rubocop/cop/rspec/unspecified_exception.rb

```diff
@@ -0,0 +1,79 @@
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks for a specified error in checking raised errors.
+      #
+      # Enforces one of an Exception type, a string, or a regular
+      # expression to match against the exception message as a parameter
+      # to `raise_error`
+      #
+      # @example
+      #
+      #     # bad
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error
+      #
+      #     # good
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError)
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError) { |err| expect(err.data).to eq(42) }
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError) do |err|
+      #       expect(err.data).to eq(42)
+      #     end
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error('error')
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError, 'error')
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(/err/)
+      #
+      #     expect {
+      #       raise StandardError.new('error')
+      #     }.to raise_error(StandardError, /err/)
+      #
+      #     expect { do_something }.not_to raise_error
+      class UnspecifiedException < Cop
+        MSG = 'Specify the exception being captured'.freeze
+
+        def_node_matcher :empty_raise_error_or_exception_call, <<-PATTERN.freeze
+          (send
+            (block
+                (send nil? :expect) ...)
+            :to
+            (send nil? {:raise_error :raise_exception})
+          )
+        PATTERN
+
+        def on_send(node)
+          return unless empty_raise_error_or_exception_call(node) &&
```

Comment:
I wonder if this can be named shorter, so it fits one line

Reply:
it can!